Microsoft.Toolkit.Uwp.UI/Shadows/AttachedShadowElementContext.cs

```diff
@@ -231,7 +234,7 @@ public T GetResource<T>(string key)
         /// <returns>The resource that was removed, if any</returns>
         public T RemoveResource<T>(string key)
         {
-            if (_resources.TryGetValue(key, out var objResource))
+            if (_resources != null && _resources.TryGetValue(key, out var objResource))
```

Comment:
this can also be `_resources.TryGetValue(key, out var objResource) == true` but not sure which is more efficient -  @Sergio0694?

Reply:
If `_resources` is nullable, you'd still want the null check or it'll throw.

This seems fine as is since it's pretty readable, but you could also do:
```csharp
if (_resources?.TryGetValue(key, out var objResource) ?? false)
```

Reply:
I mistyped and forgot the `?.` initially in my comment, I've edited it. Just trying to avoid the double-reference to `_resources`. I did see the IL was different, but not familiar enough to know the impact. The `?? false` is an interesting one but would assume the `== true` is more efficient? Lots of interesting questions here. :)

Reply:
I don't think such a micro-optimization would have any impact here given what the rest of the code is doing. We can just use whatever syntax is more convenient. Personally, doing the explicit check first (use `is not null` though) and then getting the value has the advantage that it lets the compiler infer that the value is not null if the call succeeds (assuming that API is annotated, which it should at least on WinUI 3).

Reply:
Updated :)